fix: Bacalhau roadmap yields internal server error

**Error description:** 
- Searching for the Bacalhau roadmap in starmaps results in internal server error (https://www.starmaps.app/roadmap/github.com/filecoin-project/bacalhau/issues/1151).
<img width="567" alt="Screen Shot 2022-11-21 at 11 13 24 AM" src="https://user-images.githubusercontent.com/58958327/203139863-bece3985-defd-44ae-a621-4cbc89fe4795.png">

**Addt'l Context:** 
- This error was discovered in chrome, safari, and brave
- All issues were rendering correctly in GitHub as recently as 3pm PT Sunday, Nov 20.  Issue maintainer first noticed the bug the evening of Sunday, Nov 20

**Fix**
- Not sure. Some Bacalhau child milestones yield the same error (e.g. https://www.starmaps.app/roadmap/github.com/filecoin-project/bacalhau/issues/1220) while others don't (e.g. https://www.starmaps.app/roadmap/github.com/filecoin-project/bacalhau/issues/1226). These two issues are structured exactly the same and carry the same label assignments, so I can't discern the cause

errors from vercel show:

```bash
[GET] /api/roadmap?owner=filecoin-project&repo=bacalhau&issue_number=1151&filter_group=undefined
17:05:40:35
info  - Loaded env from /var/task/.env.production
API hit: roadmap {
  repo: 'bacalhau',
  issue_number: '1151',
  filter_group: 'undefined',
  owner: 'filecoin-project'
}
2022-11-22T01:05:47.105Z	84953bff-dc86-4e36-8882-2a518f501e96	ERROR	error: Error: Unsupported URL
    at paramsFromUrl (/var/task/.next/server/pages/api/roadmap.js:380:15)
    at /var/task/.next/server/pages/api/roadmap.js:493:101
    at Array.map (<anonymous>)
    at resolveChildren (/var/task/.next/server/pages/api/roadmap.js:492:33)
    at /var/task/.next/server/pages/api/roadmap.js:508:44
    at Array.map (<anonymous>)
    at resolveChildrenWithDepth (/var/task/.next/server/pages/api/roadmap.js:506:41)
    at async /var/task/.next/server/pages/api/roadmap.js:512:27
    at async Promise.all (index 1)
    at async resolveChildrenWithDepth (/var/task/.next/server/pages/api/roadmap.js:506:16)
2022-11-22T01:05:51.931Z	84953bff-dc86-4e36-8882-2a518f501e96	WARN	API response for /api/roadmap?repo=bacalhau&issue_number=1151&filter_group=undefined&owner=filecoin-project exceeds 4MB. API Routes are meant to respond quickly. https://nextjs.org/docs/messages/api-routes-response-size-limit
[ERROR] [1669079152726] LAMBDA_RUNTIME Failed to post handler success response. Http response code: 413.
[GET] /roadmap/github.com/filecoin-project/bacalhau/issues/1151
17:05:38:59
2022-11-22T01:05:52.813Z	56c127f6-e5ee-43e8-9bb0-805142a81b88	ERROR	FetchError: invalid json response body at https://starmaps-53a2zomc1-ipfs-ignite.vercel.app/api/roadmap?owner=filecoin-project&repo=bacalhau&issue_number=1151&filter_group=undefined reason: Unexpected token < in JSON at position 0
    at /var/task/node_modules/next/dist/compiled/node-fetch/index.js:1:51220
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getServerSideProps (/var/task/.next/server/pages/roadmap/[...slug].js:1828:22)
    at async Object.renderToHTML (/var/task/node_modules/next/dist/server/render.js:508:20)
    at async doRender (/var/task/node_modules/next/dist/server/base-server.js:687:38)
    at async cacheEntry.responseCache.get.isManualRevalidate.isManualRevalidate (/var/task/node_modules/next/dist/server/base-server.js:796:28)
    at async /var/task/node_modules/next/dist/server/response-cache/index.js:80:36 {
  type: 'invalid-json',
  page: '/roadmap/[...slug]'
}
RequestId: 56c127f6-e5ee-43e8-9bb0-805142a81b88 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

--- a/components/roadmap-grid/RoadmapTabbedView.tsx
+++ b/components/roadmap-grid/RoadmapTabbedView.tsx
@@ -6,6 +6,8 @@ import {
   TabPanels,
   Tabs
 } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+
 import { setViewMode, useViewMode } from '../../hooks/useViewMode';
 import { DEFAULT_INITIAL_VIEW_MODE } from '../../lib/defaults';
 import { ViewMode } from '../../lib/enums';
@@ -16,6 +18,7 @@ import { RoadmapDetailed } from './RoadmapDetailedView';
 
 export function RoadmapTabbedView({ issueData }: { issueData: IssueData; }) {
   const viewMode = useViewMode() || DEFAULT_INITIAL_VIEW_MODE;
+  const router = useRouter();
   // Defining what tabs to show and in what order
   const tabs = ['Overview', 'Detailed View'] as const;
 
@@ -35,6 +38,9 @@ export function RoadmapTabbedView({ issueData }: { issueData: IssueData; }) {
 
   const handleTabChange = (index: number) => {
     setViewMode(tabViewMap[tabs[index]]);
+    router.push({
+      hash: tabViewMap[tabs[index]],
+    }, undefined, { shallow: true });
   }
 
   const renderTab = (title: string, index: number) => (

--- a/lib/backend/errorManager.ts
+++ b/lib/backend/errorManager.ts
@@ -42,7 +42,7 @@ export class ErrorManager {
     errorMessage,
     userGuideSection
   }: {
-    issue: GithubIssueData;
+    issue: Pick<GithubIssueData, 'html_url' | 'title'>;
     errorTitle: string;
     errorMessage: string;
     userGuideSection: string;

--- a/lib/paramsFromUrl.ts
+++ b/lib/paramsFromUrl.ts
@@ -12,8 +12,8 @@ export function paramsFromUrl (urlString: string): UrlMatchSlugs {
     if (matchResult !== false && matchResult.params != null) {
       return { ...matchResult.params };
     }
-    throw new Error('Could not parse URL');
+    throw new Error(`Could not parse URL: ${urlString}`);
   } catch {
-    throw new Error('Unsupported URL');
+    throw new Error(`Unsupported URL: ${urlString}`);
   }
 };

--- a/lib/paramsFromUrl.ts
+++ b/lib/paramsFromUrl.ts
@@ -3,6 +3,9 @@ import { slugsFromUrl } from './slugsFromUrl';
 import { UrlMatchSlugs } from './types';
 
 export function paramsFromUrl (urlString: string): UrlMatchSlugs {
+  if (urlString.includes('/issues/') && urlString.lastIndexOf('#') > 0) {
+    urlString = urlString.split('#')[0];
+  }
   try {
     const urlObj = getValidUrlFromInput(urlString)
     const matchResult = slugsFromUrl(urlObj.pathname)

--- a/pages/api/roadmap.ts
+++ b/pages/api/roadmap.ts
@@ -21,15 +21,30 @@ async function resolveChildren (children: ParserGetChildrenResponse[]): Promise<
   }
 
   return Promise.all(children.map(async (child): Promise<GithubIssueDataWithGroup> => {
-    const urlParams = paramsFromUrl(child.html_url);
-    const issueData = await getIssue(urlParams);
-    checkForLabel(issueData);
-    return {
-      ...issueData,
-      labels: issueData?.labels ?? [],
-      group: child.group
-    };
-  }));
+    try {
+      const urlParams = paramsFromUrl(child.html_url);
+      const issueData = await getIssue(urlParams);
+      checkForLabel(issueData);
+      return {
+        ...issueData,
+        labels: issueData?.labels ?? [],
+        group: child.group
+      };
+    } catch (err) {
+      errorManager.addError({
+        issue: {
+          html_url: child.html_url,
+          title: child.html_url,
+        },
+        errorTitle: 'Error parsing issue',
+        errorMessage: (err as Error).message,
+        userGuideSection: '#children'
+      })
+      throw new Error(`Error parsing issue: ${err}`);
+    }
+  })).catch((reason) => {
+    throw new Error(`Error resolving children: ${reason}`);
+  });
 };
 
 async function resolveChildrenWithDepth(children: ParserGetChildrenResponse[]): Promise<GithubIssueDataWithGroupAndChildren[]> {

--- a/pages/roadmap/[...slug].tsx
+++ b/pages/roadmap/[...slug].tsx
@@ -8,10 +8,15 @@ import NewRoadmap from '../../components/roadmap/NewRoadmap';
 import { API_URL } from '../../config/constants';
 import { IssueData, RoadmapApiResponse, RoadmapApiResponseFailure, RoadmapApiResponseSuccess, ServerSidePropsResult } from '../../lib/types';
 import { ErrorNotificationDisplay } from '../../components/errors/ErrorNotificationDisplay';
+import { ViewMode } from '../../lib/enums';
+import { setViewMode } from '../../hooks/useViewMode';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
 
 export async function getServerSideProps(context): Promise<ServerSidePropsResult> {
   const [hostname, owner, repo, issues_placeholder, issue_number] = context.query.slug;
-  const { filter_group, mode } = context.query;
+  const { filter_group, mode, view } = context.query;
+
   const res = await fetch(
     new URL(`${API_URL}?owner=${owner}&repo=${repo}&issue_number=${issue_number}&filter_group=${filter_group}`),
   );
@@ -31,6 +36,13 @@ export async function getServerSideProps(context): Promise<ServerSidePropsResult
 
 export default function RoadmapPage(props: InferGetServerSidePropsType<typeof getServerSideProps>) {
   const { issueData, error, errors, isLocal, mode } = props;
+
+  const router = useRouter();
+  const urlPath = router.asPath
+  useEffect(() => {
+    const hashString = urlPath.split('#')[1] as ViewMode ?? ViewMode.Simple;
+    setViewMode(hashString);
+  }, [urlPath])
 
   return (
     <>

--- a/tests/unit/paramsFromUrl.test.ts
+++ b/tests/unit/paramsFromUrl.test.ts
@@ -32,4 +32,12 @@ describe('paramsFromUrl', function() {
       issue_number: '1',
     });
   });
+
+  it('Should not throw an error for a url with hashString', () => {
+    expect(paramsFromUrl('https://github.com/ipfs/kubo/issues/6523#issuecomment-547254760')).toEqual({
+      owner: 'ipfs',
+      repo: 'kubo',
+      issue_number: '6523',
+    });
+  })
 })

